extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/Algorithm.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.elytron.security.runtime;
+
+import org.wildfly.security.password.interfaces.DigestPassword;
+
+/**
+ * Enum to map UserInput Algorithm Values to wildfly Algorithm
+ * Ex: ALGORITHM_DIGEST_MD5 -> digest-md5
+ */
+public enum Algorithm {
+    ALGORITHM_DIGEST_MD5(DigestPassword.ALGORITHM_DIGEST_MD5),
+    ALGORITHM_DIGEST_SHA(DigestPassword.ALGORITHM_DIGEST_SHA),
+    ALGORITHM_DIGEST_SHA_256(DigestPassword.ALGORITHM_DIGEST_SHA_256),
+    ALGORITHM_DIGEST_SHA_384(DigestPassword.ALGORITHM_DIGEST_SHA_384),
+    ALGORITHM_DIGEST_SHA_512(DigestPassword.ALGORITHM_DIGEST_SHA_512),
+    ALGORITHM_DIGEST_SHA_512_256(DigestPassword.ALGORITHM_DIGEST_SHA_512_256);
+
+    private final String algorithm;
+
+    Algorithm(String algorithm) {
+        this.algorithm = algorithm;
+    }
+
+    public static String getValue(String name) {
+        try {
+            return Algorithm.valueOf(name).algorithm;
+        } catch (IllegalArgumentException e) {
+            return name;
```

Comment:
The point of having an enum is to enforce that the random value is not entered, returning the invalid value here again makes the use of enum in this case redundant.

Reply:
Hey @sberyozkin  I was under the impression that we are giving user flexibility to enter any value and if that value is invalid then elytron will throw error and we added enum so that user can enter either ALGORITHM_DIGEST_MD5 or digest-md5. Please suggest how should we implement this.

Reply:
Hi @lower-case, I did not suggest to have multiple values for a single algorithm be supported.

Your original `geAlgorithm()` implementation was defaulting to some value the user entered entered something unrecognized, which was incorrect IMHO.

Next I proposed to throw the exception. IMHO it is better to report a Quarkus level error such as `"Algorithm %s" + algorithm is not supported, only this, that, and that algorithm values are allowed`, instead of letting the internal implementation exception be reported (by the way, how would it look if you add an algorithm value `a` ?)

Your follow up update made `getAlgorithm` redundant as you were simply passing the unrecognised value to Elytron, which does not look right either, so next I proposed to use an enum to restrict the values.

`ALGORITHM_DIGEST_MD5` is not really a user friendly value, is an internal constant, `digest-md5` is. So I proposed to have an enum which accepts `digest-md5` and other values known to the users. I believe you can add an MP Config `ConfigConverter` which will create this enum out  of `digest-md5`/etc, and make sure that the `ConfigurationException` is reported early. Or you can have enum values like `digest-md5` and don't do anything - but the exception in case of the unrecognized values won't be too useful, so the converter approach seems to be the right one

Thanks. CC @geoand @gsmet - hi, if you think I'm over complicating then please suggest @lower-case a simpler path :-)

Reply:
Spot on @sberyozkin

Reply:
@geoand thanks for checking it. @lower-case, in `quarkus-oidc` we use enums a lot, we do not use `ConfigConverter` but we don't need to map to some internal values either. I also spotted a mistake in my proposal, `-` is not a valid enum name character, so one can't have 1 one to one mapping with the enum name like `digest-md5`, hence the converter does seem the best option, it will will take care of converting a user supplied `digest-md5` into the enum value.
If it proves too tricky then may be return to the original proposal where you check the String algorithm value manually and report an exception. thanks